bin/datasource-invoke.js

```diff
@@ -68,12 +74,19 @@ function invoke(msg, cb) {
   }
 }
 
+// XXX(sam) don't do this, its unnecessary and dangerous, its not guaranteed
+// by node that this is valid, and the default behaviour of writing to stderr
+// is perfect for you, since you already collect the stderr in the parent.
 process.on('uncaughtException', function(err) {
+  // XXX(sam) the only way this is reachable is if globalize fails to
+  // initialize, AND this process doesn't have an IPC channel... delete all this
+  // code along with the uncaught exception handler.
```

Comment:
IIRC, the idea here is that if the LoopBack app cannot be loaded because the connection configuration of a datasource is invalid, we forward the error message to the caller, so that the caller can provide a specific error code for the UI, so that UI can show a useful error message like "cannot connect to the database" instead of a generic "datasource-invoke crashed".

Having said that, I think parsing the stdout/stderr output to extract the error message from there is a viable strategy too. Especially if the current implementation is not reliable enough.